subt_ign/launch/competition.ign

```diff
@@ -42,13 +91,80 @@
+        robots[name] = Robot.new(name, config)
       end
     end
   end
+
+  MARSUPIAL_VALID_PARENT_ROBOT_TYPES = ["X1"]
+  MARSUPIAL_VALID_CHILD_ROBOT_TYPES = ["X3"]
+
+  MARSUPIAL_PARENT_ROBOT_POSITION_OFFSETS = {
+    "X1" => Vector3d.new(0, 0, 0.53)
+  }
+
+  allExecutables = ""
+
+  # key: parent, value: child
+  marsupialParents = Hash.new
+
+  # The logic for validating marsupial robots is:
+  # * Check that parent names are unique
+  # * Check that child names are unique
+  # * Check that each parent and child robot is in the list of robots to be spawned
+  # * Check that parent robots are in the MARSUPIAL_VALID_PARENT_ROBOT_TYPES list
+  # * Check that child robots are in the MARSUPIAL_VALID_CHILD_ROBOT_TYPES list
+  # These criteria ensure that the robots specified in the "marsupial" variables are valid
+  # and that there is a one-to-one relationship between parent and child robots
+
+  # Handle marsupial robots
+  for i in 1..maxRobotCount do
+    if local_variables.include?(:"marsupial#{i}")
+      parent, child = (eval "marsupial#{i}").split(':')
+
+      # Check that parent names are unique
+      if marsupialParents.has_key?(parent)
+        raise "Invalid marsupial configuration: The parent robot [#{parent}] cannot be used more than once"
+      end
+
+      # Check that child names are unique
+      if marsupialParents.has_value?(child)
+        raise "Invalid marsupial configuration: The child robot [#{child}] cannot be used more than once"
+      end
+
+      # Check that each parent and child robot is in the list of robots to be spawned
+      unless robots.has_key?(parent)
+        raise "Invalid marsupial configuration: The parent robot [#{parent}] is not in the list of robots to be spawned"
+      end
+      unless robots.has_key?(child)
+        raise "Invalid marsupial configuration: The child robot [#{child}] is not in the list of robots to be spawned"
+      end
+
+      parentType = robots[parent].type
+      childType = robots[child].type
+      # Check that parent robots are in the MARSUPIAL_VALID_PARENT_ROBOT_TYPES list
+      unless MARSUPIAL_VALID_PARENT_ROBOT_TYPES.include?(parentType)
+        raise "Invalid marsupial configuration: The parent robot [#{parent}] with type [#{parentType}] is not in the list of robots allowed to be marsupial parents. The list is #{MARSUPIAL_VALID_PARENT_ROBOT_TYPES}"
+      end
+      # Check that child robots are in the MARSUPIAL_VALID_CHILD_ROBOT_TYPES list
+      unless MARSUPIAL_VALID_CHILD_ROBOT_TYPES.include?(childType)
+        raise "Invalid marsupial configuration: The child robot [#{child}] with type [#{childType}] is not in the list of robots allowed to be marsupial children: The list is #{MARSUPIAL_VALID_CHILD_ROBOT_TYPES}"
+      end
+
+      marsupialParents[parent] = child
+    end
+  end
 %>
 
 <%
+  # Check if worldName is not defined or is empty/nil
+  if !defined?(worldName) || worldName == nil || worldName.empty?
+    $worldName = 'simple_cave_01'
```

Comment:
Should this be tunnel_qual?

Reply:
It's okay to update this so that it points to the most recent circuit.